csharp-client-reference/MainWindow.xaml.cs

```diff
@@ -88,12 +88,17 @@ private async Task HandlingRequestsAsync(AsyncServerStreamingCall<OASISQueryDTO>
                 {
                     HandleRequestAsync(requestsResponseStream.Current);
                 }
-                Log("- Query Stream Closed, unregister");
-                Unregister();
+
+                Log("- Query Closed, plugin has been unregisred by server");
```

Comment:
Now, when a node register failed or being unregistered from server, the server will call complete on the StreamObserver and `while (await requestsResponseStream.MoveNext(new CancellationToken()))` will finish. At this point, client should consider itself unregistered.